server/controllers/Project.js

```diff
@@ -0,0 +1,96 @@
+const { User, Project } = require('../models');
+const formidable = require('formidable');
+const upload = require('../services/file-upload');
+const singleUpload = upload.single('image');
+const multiUpload = upload.array('images');
+
+
+exports.imageUpload = (req, res) => {
+    singleUpload(req, res, function (err) {
+        if (err) return res.status(422).json({ errors: [{ title: 'File Upload Error', detail: err.message }] });
+        return res.json({ 'imageUrl': req.file.location })
+    })
+}
+
+exports.projectById = (req, res, next, id) => {
+    Project.findById(id)
+        .populate('user')
+        .exec((err, project) => {
+            if (err || !project) {
+                return res.status(400).json({
+                    error: 'Project not found.'
+                });
+            }
+            req.project = project;
+            next();
+        })
+}
+
+exports.getUserProjects = (req, res) => {
+    // console.log(req.profile._id)
+    // Project.find({ user: req.profile._id })
+    //     .exec((err, projects) => {
+    //         if (err) {
+    //             return res.status(400).json({
+    //                 error: 'Projects not found.'
+    //             });
+    //         }
+    //         res.json(projects)
+    //     })
```

Comment:
Probably should remove this if not being used

Reply:
I left it because it was the initial way I had wanted to do it but then Shums mentioned the populate method. However, I'm not sure if this is actually what he wanted (although this does give use all the projects associated with the user). I'm going to clean it up after stand up for another review before merging